lib/matplotlib/table.py

```diff
@@ -838,5 +838,9 @@ def table(ax,
         if rowLabelWidth == 0:
             table.auto_set_column_width(-1)
 
+    # set_fontsize is only effective after cells are added
+    if fontsize is not None:
+        table.set_fontsize(fontsize)
```

Comment:
Optional: As an alternative to changing the spelling of our signatures, we could get fontsize from kwargs here. That way, it doesn't show up in the signature explicitly (which may be an advantage or disavantage depending on your perspective.

```suggestion
    if "fontsize" in kwargs:
        table.set_fontsize(kwargs["fontsize"])
```